module/VuFind/src/VuFind/Recommend/LibGuidesProfile.php

```diff
@@ -0,0 +1,241 @@
+     *
+     * @var array
+     */
+    protected $subjectToId = [];
+
+    /**
+     * LibGuides connector
+     *
+     * @var LibGuides
+     */
+    protected $libGuides;
+
+    /**
+     * Constructor
+     *
+     * @param array      $config Config object representing LibGuidesAPI.ini
+     * @param HttpClient $client VuFind HTTP client
+     */
+    public function __construct($config, HttpClient $client)
+    {
+        $this->libGuides = new LibGuides(
+            $client,
+            $config->General->api_base_url,
+            $config->General->client_id,
+            $config->General->client_secret
+        );
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search.  This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->results = $results;
+    }
+
+    /**
+     * Get terms related to the query.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        // TODO: cache data, check it for staleness
+        $this->refreshData();
+
+        $query = $this->results->getParams()->getQuery();
+        $account = $this->findBestMatch($query);
+        return $account;
+    }
+
+    /**
+     * Find the VuFind account whose profile best matches the
+     * given query.
+     *
+     * @param \VuFindSearch\Query\QueryInterface $query Current search query
+     *
+     * @return array LibGuides account
+     */
+    protected function findBestMatch(\VuFindSearch\Query\QueryInterface $query)
+    {
+        $queryString = $query->getAllTerms();
+        if (!$queryString) {
+            return false;
+        }
+        $queryString = strtolower($queryString);
+
+        // Find the closest levenshtein match.
```

Comment:
See comments in the PR description.  I recognize it's weird in a project reliant on Solr and its excellent searching and relevancy ranking to do something like this algorithm.  But it works surprisingly well, and fast enough in my own testing.  Thoughts?

Reply:
The thing about levenshtein is it often works well even if they do type in something really long & complex, like "new approaches in neurobiology" would pull up a librarian with a "biology" guide over a humanities librarian, etc.    I agree through that using the result set's call number facets is a good idea if someone is willing to maintain the local configuration as Villanova did, and if the primary search will return call numbers (so for an ILS definitely, although not EDS or such) -- I added that as a TODO on the PR.

The tougher problem (IMHO) with levneshtein as you reference is that it's impossible by itself to know if you have no good matches at all -- the result distance just tells you better and worse, with no useful threshold for "good enough".  So no concept of "very generalized searches" that's detectable, or trolling searches like "show me the profile of a horrible librarian".  I discussed this with one subject librarian and she thought it was still acceptable -- assuming good intent by the patron, if they get the wrong librarian and send them a query, the librarian will just forward it on to the appropriate person; better to still show the patron that a librarian can help them.